internal/pkg/service/buffer/api/service/service.go

```diff
@@ -64,14 +65,28 @@ func (s *service) CreateReceiver(d dependencies.ForProjectRequest, payload *buff
 	// Generate Secret
 	secret := idgenerator.ReceiverSecret()
 
+	// Create Storage files for exports
+	storageClient := d.StorageAPIClient()
+	files := make([]*storageapi.File, 0)
+	for _, export := range payload.Exports {
+		fileRes, err := storageapi.CreateFileResourceRequest(&storageapi.File{
+			Name:     export.Name,
+			IsSliced: true,
+		}).Send(ctx, storageClient)
+		if err != nil {
+			return nil, errors.Errorf("creating Storage file failed: %w", err)
+		}
+		files = append(files, fileRes)
```

Comment:
Toto by bolo dobre spravit paralelne, ak si niekto definuje maximalny pocet exportov (`20`), tak toto sekvencne vytvaranie moze sposobit timeout requestu na nase API. 

Pozri `WaitGroup` v `go-client`.

Reply:
Asi tam bude potrebny aj jednoduchy zamok na pridavanie fo `files`, `&sync.Mutex`.

Reply:
Dalo by sa to vyriesit, tak ako to spravil Honza tu, ... tj. bez zamku, zapis na konkretny index slice:
https://github.com/keboola/keboola-as-code/blob/be361bddb7c8f8be3b30137ca30dc0d1c3a57f81/internal/pkg/service/buffer/api/service/service.go#L182-L201

Reply:
Ale nakoniec by to mala byt asi `map` ako pisem nizsie a teda bude tam potrebny aj zamok.

Reply:
Omg jasně. 🤦‍♂️ to jsem tam jen dal narychlo na začátku a zapomněl se k tomu vrátit. 😅